rust-src/id/src/ffi.rs

```diff
@@ -292,6 +298,203 @@ pub extern "C" fn ip_info_cdi_verify_key(
     ptr
 }
 
+#[no_mangle]
+/// Instantiate an ArInfo instance from byte arrays.
+unsafe extern "C" fn ar_info_create(
+    identity: u32,
+    public_key_ptr: *mut u8,
+    public_key_len: size_t,
+    name_ptr: *const u8,
+    name_len: size_t,
+    url_ptr: *const u8,
+    url_len: size_t,
+    desc_ptr: *const u8,
+    desc_len: size_t,
+) -> *mut ArInfo<G1> {
+    // Description.
+    let name;
+    if let Ok(_name) = from_utf8(slice_from_c_bytes!(name_ptr, name_len as usize)) {
+        name = _name.to_string();
+    } else {
+        return std::ptr::null_mut();
+    }
```

Comment:
You should just state a precondition to this function and not handle these problems here. This is an internal function and we know it will always be called with valid utf8 strings.